pkg/kubelet/probe.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/probe"
+	execprobe "github.com/GoogleCloudPlatform/kubernetes/pkg/probe/exec"
+	httprobe "github.com/GoogleCloudPlatform/kubernetes/pkg/probe/http"
+	tcprobe "github.com/GoogleCloudPlatform/kubernetes/pkg/probe/tcp"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
+
+	"github.com/golang/glog"
+)
+
+func (kl *Kubelet) makeLivenessProbeRunner(podFullName string, podUID types.UID, status api.PodStatus, container api.Container) probe.ProbeRunner {
```

Comment:
I'd just pass in LivenessProbe and any other information from the container needed (e.g., name). That would make it easier to reuse this for readiness probes.

We should change the name of type LivenessProbe to Probe.